test/release/examples/patterns/readcsv.chpl

```diff
@@ -0,0 +1,286 @@
+
+  //-----------------------------------------------------------------------
+  // To see how the data is accessed and for testing, let's generate a 
+  // temporary csv file from the stored data.  It should be the same as the 
+  // input csv file.
+  if debug {
+    var outfile = open(approach2FileName, iomode.cw);
+    var writer = outfile.writer();
+
+    // First write to the output file the column names separated by commas.
+    for colIdx in 0..colNames.size-2 {
+      writer.write(colNames[colIdx],",");
+    }
+    // The last column name should not be followed by comma.
+    writer.writeln(colNames[colNames.size-1]);  
+
+    // Then write to the file each row of data.
+    for rowIdx in 0..colData[colNames[0]].size-1 {
+      for colIdx in 0..colNames.size-2 {
+        writer.write(colData[colNames[colIdx]][rowIdx],",");
+      }
+      // The last column value should not be followed by comma.
+      writer.writeln(colData[colNames[colNames.size-1]][rowIdx]);
+    }
+    writer.close();
+
+    // Compare the file generated from the associative array with the original input csv file.
+    checkThatFilesMatch(inFileName,approach2FileName);
+  }
+}
+
+// Returns the index of the next comma or the string length if no
+// next comma is found.  Commas inside quoted strings will be ignored.
+proc findNextCommaNotInQuotes(str : string, start : int) {
+  var inQuotes = false;
+  for i in start..(str.size-1) {
+    if !inQuotes && str[i]==',' then return i;
+    if str[i]=='"' then inQuotes = !inQuotes;
+  }
+  return str.size;
+}
+
+// Given a line from a csv file and a starting index
+// returns a tuple of the string starting at the starting index and
+// ending right before the next comma not in a quoted string, and
+// the index of the next comma.
+//      input: "ab,cd,"ef",f"   and start=3
+//      output: "cd"
+//
+//      input: "ab,cd,"ef",f"   and start=6
+//      output: ""ef""
+// Assumes that the starting index given is not within a quoted string.
+proc nextField(line:string, start:int) {
+    var commaIdx = findNextCommaNotInQuotes(line,start);
+    var fieldVal = line[start..(commaIdx-1)];
+    return (fieldVal.strip(),commaIdx);
+}
+
+// Given the string that is the first line of a csv file that has column names
+// returns a list of column names in order.
+proc createListOfColNames(line : string) {
+  var colNames : list(string);
+  var start = 0;
+  while (start<line.size) {
+    var (nextVal,commaIdx) = nextField(line,start);
+    colNames.append(nextVal);
+    start = commaIdx+1;
+  }
+  return colNames;
+} 
+
+// A function that checks the differences between files.
+// Prints out an Error message if the files don't match and "Test Passed" 
+// if the files do match.
+proc checkThatFilesMatch(file1 : string, file2 : string) {
+  use IO;
+
+  // Open the files.
+  var f1 = open(file1, iomode.r);
+  var f2 = open(file2, iomode.r);
+
+  // read in all of the bytes from each file into a string
+  var file1data : string;
```

Comment:
Capitalize as `file1Data` ? (Same for `file2Data`)

Reply:
done